app/vmalert/static/js/custom.js

```diff
@@ -83,81 +65,51 @@ function getParamURL(key) {
     return url.searchParams.get(key)
 }
 
-function filterGroupsByName(searchPhrase) {
-    $(".group-heading").each(function () {
-        const groupName = $(this).attr('data-group-name').toLowerCase();
-        const hasValue = groupName.indexOf(searchPhrase) >= 0
-
-        if (!hasValue) {
-            return
-        }
-
-        const target = $(this).attr("data-bs-target");
-        $(`div[id="${target}"] .rule`).show();
-        $(this).show();
-    });
-}
-
-function filterRuleByName(searchPhrase) {
-    $(".rule").each(function () {
-        const ruleName = $(this).attr("data-rule-name").toLowerCase();
-        const hasValue = ruleName.indexOf(searchPhrase) >= 0
-        if (!hasValue) {
-            $(this).hide();
-            return
-        }
-
-        const target = $(this).attr('data-bs-target')
-        $(`#rules-${target}`).addClass('show');
-        $(`div[data-bs-target='rules-${target}']`).show();
-        $(this).show();
-    });
-}
-
-function filterRuleByLabels(searchPhrase) {
-    $(".rule").each(function () {
-        const matches = $(".label", this).filter(function () {
-            const label = $(this).text().toLowerCase();
-            return label.indexOf(searchPhrase) >= 0;
-        }).length;
-
-        if (matches > 0) {
-            const target = $(this).attr('data-bs-target')
-            $(`#rules-${target}`).addClass('show');
-            $(`div[data-bs-target='rules-${target}']`).show();
-            $(this).show();
+function filterRules(searchPhrase) {
```

Comment:
I'm not sure if it was like that before, but the current behavior is following:
1. Group is expanded and all rules are renderred
2. Start searching for a rule
3. Delete the search filter
4. All rules are hidden

Can we fix p4 so all rules remain visible as before searching?

Reply:
i cannot reproduce this, this case was covered

Reply:
https://github.com/user-attachments/assets/b10989d1-b9f5-4011-9dcf-9320a2d4ddc3

Reply:
fixed it, but removed automatic expansion of search results

Reply:
Not ideal, but ok. Let's see later if that needs to be improved